src/Cache/MongoLock.php

```diff
@@ -74,11 +74,11 @@ public function acquire(): bool
             ],
         );
 
-        if (random_int(1, $this->lottery[1]) <= $this->lottery[0]) {
-            $this->collection->deleteMany(['expiration' => ['$lte' => $this->currentTime()]]);
+        if (! empty($this->lottery[0]) && random_int(1, $this->lottery[1]) <= $this->lottery[0]) {
```

Comment:
Does anything enforce that `$this->lottery` is a tuple? This only checks that the first element exists and is non-empty, but there is no check for a second element or both being integers.

Perhaps that should be done in the constructor.

I assume you intentionally decided not to make `$lottery` nullable as a means to disable it.

Reply:
It seemed to me that the "**zero** chances on **whatever**" value was explicit enough to say that we wanted to disable this mechanism.